monai/networks/blocks/denseblock.py

```diff
@@ -129,3 +129,178 @@ def _get_layer(self, in_channels, out_channels, dilation):
+        seLayer: Squeeze and Excite block to be included, defaults to None, valid options are {'NONE', 'CSE', 'SSE', 'CSSE'},
+        dropout_layer: Dropout block to be included, defaults to None.
+    :return: forward passed tensor
+    """
+
+    def __init__(
+        self,
+        in_channels: int,
+        se_layer: nn.Module | None = None,
+        dropout_layer: type[nn.Dropout2d] | None = None,
+        kernel_size: Sequence[int] | int = 5,
+        num_filters: int = 64,
+    ):
+        self.count = 0
+        super().__init__(
+            in_channels=in_channels,
+            spatial_dims=2,
+            # number of channels stay constant throughout the convolution layers
+            channels=[num_filters, num_filters, num_filters],
+            norm=("instance", {"num_features": in_channels}),
+            kernel_size=kernel_size,
+        )
+        self.se_layer = se_layer
+        self.dropout_layer = dropout_layer
+
+    def _get_layer(self, in_channels, out_channels, dilation):
+        """
+        After ever convolutional layer the output is concatenated with the input and the layer before.
+        The concatenated output is used as input to the next convolutional layer.
+
+        Args:
+            in_channels: number of input channels.
+            out_channels: number of output channels.
+            strides: convolution stride.
+            is_top: True if this is the top block.
+        """
+        kernelsize = self.kernel_size if self.count < 2 else (1, 1)
+        # padding = None if self.count < 2 else (0, 0)
+        self.count += 1
+        conv = Convolution(
+            spatial_dims=self.spatial_dims,
+            in_channels=in_channels,
+            out_channels=out_channels,
+            strides=1,
+            kernel_size=kernelsize,
+            act=self.act,
+            norm=("instance", {"num_features": in_channels}),
+        )
+        return nn.Sequential(conv.get_submodule("adn"), conv.get_submodule("conv"))
+
+    def forward(self, input, _):
+        i = 0
+        result = input
+        for l in self.children():
+            # ignoring the max (un-)pool and droupout already added in the initial initialization step
+            if isinstance(l, nn.MaxPool2d) or isinstance(l, nn.MaxUnpool2d) or isinstance(l, nn.Dropout2d):
+                continue
+            # first convolutional forward
+            result = l(result)
+            if i == 0:
+                result1 = result
+                # concatenation with the input feature map
+                result = torch.cat((input, result), dim=1)
+
+            if i == 1:
+                # concatenation with input feature map and feature map from first convolution
+                result = torch.cat((result1, result, input), dim=1)
+            i = i + 1
+
+        # if SELayer or Dropout layer defined put output through layer before returning
+        if self.se_layer is not None:
```

Comment:
I would suggest setting `se_layer` and `dropout_layer` to `nn.Identity` in the constructor if the given argument is `None` rather than checking here, I think it's more compatible with Torchscript that way.